### PR TITLE
Refs [TASK][39486] Xem chi tiết tin tức

### PR DESCRIPTION
**Pull request by okuloj**
## Related Tickets
- [#Task 39486: Xem chi tiết tin tức](https://edu-redmine.sun-asterisk.vn/issues/39486)
## WHAT
- Create UI, gán dữ liệu, sự kiện click item
## Evidence (Screenshot or Video)
https://user-images.githubusercontent.com/55242770/130569918-f45f852f-d1e9-4c32-8cda-538accb54d07.mp4
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt [hunk 1]
```diff
@@ -14,6 +16,7 @@ import java.time.format.DateTimeFormatter
 class ArticlesAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
     private val articles = mutableListOf<Article?>()
+    private var onItemClickListener: OnItemRecyclerViewClickListener<Article>? = null
```
Comment (luongth-2520) [inline app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt]: suggest high order function

File: app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt [hunk 2]
```diff
@@ -14,6 +16,7 @@ import java.time.format.DateTimeFormatter
...
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt]: sử dụng higher order function cho gọn hơn

File: app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt [hunk 3]
```diff
@@ -35,11 +38,17 @@ class ArticlesAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
         }
     }
 
+    fun registerItemRecyclerViewClickListener(
+        onItemRecyclerViewClickListener: OnItemRecyclerViewClickListener<Article>?
+    ) {
+        onItemClickListener = onItemRecyclerViewClickListener
+    }
+
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt]:
```suggestion
```

File: app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt [hunk 4]
```diff
@@ -55,30 +64,42 @@ class ArticlesAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
...
 
     override fun getItemCount() = articles.size
 
-    class ArticleViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    class ArticleViewHolder(
+        view: View,
+        private val itemListener: OnItemRecyclerViewClickListener<Article>?
+    ) : RecyclerView.ViewHolder(view), View.OnClickListener {
+
+        private var listener: OnItemRecyclerViewClickListener<Article>? = null
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt]:
```suggestion
```

File: app/src/main/java/com/edu/news/screen/detail/DetailFragment.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+import android.view.View
+import androidx.core.os.bundleOf
+import com.edu.news.data.model.Article
+import com.edu.news.data.source.remote.fetchData.LoadImageFromUrl
+import com.edu.news.utils.Constant
+import com.edu.news.utils.base.BaseFragment
+import com.sun.news.R
+import kotlinx.android.synthetic.main.fragment_detail.*
+import kotlinx.android.synthetic.main.fragment_detail.view.*
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/detail/DetailFragment.kt]:
```suggestion
import kotlinx.android.synthetic.main.fragment_detail.*
```

File: app/src/main/java/com/edu/news/screen/nav/BottomNavFragment.kt [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.edu.news.screen.favorite.FavoriteFragment
+import com.edu.news.screen.follows.FollowsFragment
+import com.edu.news.screen.home.HomeFragment
+import com.edu.news.utils.MenuItem
+import com.edu.news.utils.base.BaseFragment
+import com.sun.news.R
+import kotlinx.android.synthetic.main.fragment_bottom_navigation.*
+import kotlinx.android.synthetic.main.fragment_bottom_navigation.view.*
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/nav/BottomNavFragment.kt:12]:
```suggestion
import kotlinx.android.synthetic.main.fragment_bottom_navigation.*
```

File: app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt [hunk 7]
```diff
@@ -1,17 +1,19 @@
...
 import com.edu.news.data.source.remote.fetchData.LoadImageFromUrl
+import com.edu.news.utils.Constant
 import com.sun.news.R
 import kotlinx.android.synthetic.main.item_article.view.*
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 
-class ArticlesAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class ArticlesAdapter(private val onItemClickListener: ((Article) -> Unit)) :
+    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/articles/ArticlesAdapter.kt]: format code
```suggestion
class ArticlesAdapter(
  private val onItemClickListener: ((Article) -> Unit)
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```

File: app/src/main/java/com/edu/news/screen/articles/ArticlesFragment.kt [hunk 8]
```diff
@@ -10,15 +10,21 @@ import com.edu.news.data.model.Article
...
 
 class ArticlesFragment : BaseFragment(), ArticlesContract.View {
 
     private lateinit var articlePresenter: ArticlePresenter
     private lateinit var categoryType: CategoryType
-    private val articlesAdapter by lazy { ArticlesAdapter() }
+    private val articlesAdapter by lazy {
+        ArticlesAdapter {
+            onItemClickListener(it)
+        }
```
Comment (duchh-2641) [inline app/src/main/java/com/edu/news/screen/articles/ArticlesFragment.kt]:
```suggestion
        ArticlesAdapter { onItemClickListener(it) }
```

File: app/build.gradle [hunk 9]
```diff
@@ -35,6 +35,7 @@ android {
 }
 
 dependencies {
+    implementation("androidx.fragment:fragment-ktx:1.3.6")
```
Comment (tuanlvt-1117) [inline app/build.gradle]: sao kiểu add lib này khác với những kiểu còn lại e?

File: app/src/main/java/com/edu/news/utils/Constant.kt [hunk 10]
```diff
@@ -11,8 +11,8 @@ enum class MenuItem {
 object Constant {
     const val BASE_URL = "https://newsapi.org/v2/top-headlines"
     const val BASE_COUNTRY = "?country=us"
-    const val BASE_PAGE = "&page=1"
     const val BASE_PAGE_PARAM = "&page="
     const val BASE_CATEGORY_PARAM = "&category="
     const val BASE_API_KEY = "&apikey=${BuildConfig.API_KEY}"
+    const val DATETIME_FORMAT = "dd.MM.yyyy HH:mm"
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/edu/news/utils/Constant.kt]:
```suggestion
    const val DATE_TIME_FORMAT = "dd.MM.yyyy HH:mm"
```

**Review by duchh-2641 — APPROVED** [review]

File: app/src/main/java/com/edu/news/data/model/Article.kt [hunk 11]
```diff
@@ -9,7 +13,7 @@ data class Article(
     var urlToImage: String = "",
     var publishedAt: String = "",
     var content: String = ""
-)
+) : Parcelable
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/edu/news/data/model/Article.kt:16]: thằng này là gì vậy e?

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**